komfydb/database.cc

```diff
@@ -10,129 +13,128 @@
 
 #include "komfydb/common/tuple_desc.h"
 #include "komfydb/common/type.h"
+#include "komfydb/execution/executor.h"
+#include "komfydb/execution/query.h"
+#include "komfydb/optimizer/table_stats.h"
 #include "komfydb/storage/db_file.h"
 #include "komfydb/storage/heap_file.h"
+#include "komfydb/transaction/transaction_id.h"
 #include "komfydb/utils/status_macros.h"
 
-namespace {
-
-using komfydb::common::TupleDesc;
-using komfydb::common::Type;
-using komfydb::storage::DbFile;
-using komfydb::storage::HeapFile;
-
-inline void ltrim(std::string& s) {
-  s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char ch) {
-            return !std::isspace(ch);
-          }));
-}
+namespace komfydb {
 
-// trim from end (in place)
-inline void rtrim(std::string& s) {
-  s.erase(std::find_if(s.rbegin(), s.rend(),
-                       [](unsigned char ch) { return !std::isspace(ch); })
-              .base(),
-          s.end());
+Database::Database(std::shared_ptr<Catalog> catalog,
+                   std::shared_ptr<BufferPool> buffer_pool,
+                   std::shared_ptr<Parser> parser,
+                   optimizer::TableStatsMap&& table_stats_map,
+                   std::string_view catalog_directory)
+    : catalog(catalog),
+      buffer_pool(buffer_pool),
+      parser(parser),
+      table_stats_map(std::move(table_stats_map)),
+      catalog_directory(catalog_directory) {}
+
+std::unique_ptr<Database> Database::Create(std::string_view catalog_directory) {
+  std::shared_ptr<Catalog> catalog = std::make_shared<Catalog>();
+  std::shared_ptr<BufferPool> buffer_pool =
+      std::make_shared<BufferPool>(catalog);
+  optimizer::TableStatsMap table_stats_map;
+  std::shared_ptr<Parser> parser =
+      std::make_shared<Parser>(catalog, buffer_pool, table_stats_map);
+  return std::unique_ptr<Database>(new Database(catalog, buffer_pool, parser,
+                                                std::move(table_stats_map),
+                                                catalog_directory));
 }
 
-// trim from both ends (in place)
-inline void trim(std::string& s) {
-  ltrim(s);
-  rtrim(s);
-}
+absl::Status Database::CreateTable(Query& query) {
+  if (catalog->GetTableId(query.table_name).ok()) {
+    return absl::InvalidArgumentError(
+        absl::StrCat("Table ", query.table_name, " already exists."));
+  }
+  absl::StatusOr<std::unique_ptr<DbFile>> new_file = HeapFile::Create(
+      absl::StrCat(catalog_directory, query.table_name, ".dat"),
```

Comment:
fajnie by pewnie było mieć jakąś funkcję do obsługi nazw plików z tabelkami, żeby nie trzeba było zawsze pamiętać o dodawaniu .dat itp. do nazw

Reply:
nie trzeba, samo się to robi. Robisz 'CREATE TABLE <nazwa>` i samo powstaje Ci `<nazwa>.dat`, nigdy nie tworzysz tego pliku ręcznie.